[ci] `changelog` workflow fails due to a bug in latest `markdown-link-check`

### Component(s)

_No response_

### Describe the issue you're reporting

The markdown-link-check release [v3.12.0](https://github.com/tcort/markdown-link-check/releases/tag/v3.12.0) (published on Sunday 2024-03-10 around 18:06 UTC) contains a [bug](https://github.com/tcort/markdown-link-check/issues/297) that causes it to fail if the `--ignore` flag is not passed.

Example: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8228894538/job/22505249862

```console
Run markdown-link-check \
  markdown-link-check \
    --verbose \
    --config .github/workflows/check_links_config.json \
    changelog_preview.md \
    || { echo "Check that anchor links are lowercase"; exit 1; }
  shell: /usr/bin/bash -e {0}
  env:
    SEGMENT_DOWNLOAD_TIMEOUT_MINS: 2
    PR_HEAD: 2f898337653a4b34a2b985142eec5b247dee93a5
/usr/local/lib/node_modules/markdown-link-check/markdown-link-check:97
                    if (program.opts().ignore.some((ignorePath) => resolved.includes(ignorePath))) {
                                             ^
TypeError: Cannot read properties of undefined (reading 'some')
    at Command.<anonymous> (/usr/local/lib/node_modules/markdown-link-check/markdown-link-check:97:46)
    at Command.listener [as _actionHandler] (/usr/local/lib/node_modules/markdown-link-check/node_modules/commander/lib/command.js:482:17)
    at /usr/local/lib/node_modules/markdown-link-check/node_modules/commander/lib/command.js:1283:65
    at Command._chainOrCall (/usr/local/lib/node_modules/markdown-link-check/node_modules/commander/lib/command.js:1177:12)
    at Command._parseCommand (/usr/local/lib/node_modules/markdown-link-check/node_modules/commander/lib/command.js:1283:27)
    at Command.parse (/usr/local/lib/node_modules/markdown-link-check/node_modules/commander/lib/command.js:909:10)
    at getInputs (/usr/local/lib/node_modules/markdown-link-check/markdown-link-check:108:7)
    at main (/usr/local/lib/node_modules/markdown-link-check/markdown-link-check:252:20)
Node.js v18.19.1
Check that anchor links are lowercase
Error: Process completed with exit code 1.
```

--- a/.github/workflows/changelog.yml
+++ b/.github/workflows/changelog.yml
@@ -84,7 +84,7 @@ jobs:
         run: make chlog-preview > changelog_preview.md
       - name: Install markdown-link-check
         if: ${{ !contains(github.event.pull_request.labels.*.name, 'dependencies') && !contains(github.event.pull_request.labels.*.name, 'Skip Changelog') && !contains(github.event.pull_request.title, '[chore]')}}
-        run: npm install -g markdown-link-check
+        run: npm install -g markdown-link-check@3.11.2
       - name: Run markdown-link-check
         if: ${{ !contains(github.event.pull_request.labels.*.name, 'dependencies') && !contains(github.event.pull_request.labels.*.name, 'Skip Changelog') && !contains(github.event.pull_request.title, '[chore]')}}
         run: |

